daml-foundations/daml-ghc/daml-compiler/src/DA/Service/Daml/Compiler/Impl/Scenario.hs

```diff
@@ -35,13 +30,4 @@ startScenarioService eventHandler loggerH =
           , optLogInfo = wrapLog Logger.logInfo
           , optLogError = wrapLog Logger.logError
           }
-    SS.start opts >>= \case
-      Left err -> error $ "Failed to start scenario service: " ++ show err
-      Right h -> pure h
-
-logScenarioException :: (LSP.Event -> IO ()) -> Logger.Handle IO -> IO a -> IO a
```

Comment:
This didn’t really make any sense:

1. We catch also the exceptions emitted by the callback not only those by the scenario service process so the log message is misleading.
2. Proxying this through the event logger doesn’t help with anything.
3. It actually doesn’t even work since the call to `exitFailure` in the event logger isn’t run from the main thread so it won’t tear anything down.
4. On the other hand we do run this from the main thread so if an exception bubbles up to here (which it shouldn’t), it will tear down everything anyway.